pipeline/scraper/github_query.py

```diff
@@ -87,7 +61,37 @@ def write_db(database, result):
     database.commit()
 
 
-def run_query(today):
+def write_db_python(database, result):
+    """
+    Write a set of repositories and their dependencies to the database for python packages
+    """
+
+    nodes = [edge['node'] for edge in result['data']['search']['edges']]
+    for node in nodes:
+        if not node['object']:
+            continue
+        package_str = node['object']['text']
+        packages = package_str.splitlines()
+        hash_value = hash(package_str)
+        application_id = insert_to_app(database,
+                                       node['url'],
+                                       node['watchers']['totalCount'],
+                                       node['nameWithOwner'],
+                                       hash_value)
+        for package in packages:
+            try:
+                line = package.split("==")
+                package_name = line[0]
+                package_version = line[1]
+                dependency_str = 'pkg:npm/' + package_name + "@" + package_version
+                package_id = insert_to_package(database, dependency_str)
+                insert_to_dependencies(database, str(application_id), str(package_id))
+            except:
+                continue
+    database.commit()
```

Comment:
There's a lot of code duplication here. Can you try rewriting the write_db method to support both languages, or introduce a class hierarchy with all the common functionality (e.g. parsing the response from GitHub) to the shared base class?